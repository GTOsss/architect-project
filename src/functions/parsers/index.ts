export * from './parseAssets';
export * from './parseFiles';
// export * from './parseSourceMap';
