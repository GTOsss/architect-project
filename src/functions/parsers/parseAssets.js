const { basename } = require('path');
const fs = require('file-system');
const configPath = require('../../configPath');
const getFilesPath = require('../getters/getObjectWithPaths');

const parseAssets = () => {
  const assetsPath = configPath.assetsPath;

  const allFilesPaths = getFilesPath(assetsPath);

  return allFilesPaths.reduce((acc, { templateName, files }) => {
    acc[templateName] = files.map((path) => ({
      fileName: basename(path),
      path: path,
      content: fs.readFileSync(path),
    }));
    return acc;
  }, {});
};

module.exports = parseAssets;
