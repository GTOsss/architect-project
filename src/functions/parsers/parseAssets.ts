import { basename } from 'path';
import fs from 'file-system';
import { getTemplatesInfo } from '../getters';

export const parseAssets = () => {
  const allFilesPaths = getTemplatesInfo();

  return allFilesPaths.reduce((acc, { templateName, files }) => {
    acc[templateName] = files.map((path) => ({
      fileName: basename(path),
      path: path,
      content: fs.readFileSync(path),
    }));
    return acc;
  }, {});
};
