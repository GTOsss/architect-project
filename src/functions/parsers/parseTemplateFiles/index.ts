export * from './parseTemplateFiles';
export * from './parseTemplateFiles.types';
export * from './parseTemplateFragments.types';
