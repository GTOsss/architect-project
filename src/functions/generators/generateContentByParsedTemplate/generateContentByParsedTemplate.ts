import { ParsedFragmentDataTypeEnum, ParsedTemplateFile } from '../../parsers/parseTemplateFiles';
import { SourceMapModuleConsistent, TemplateParamsConsistent } from '../../../types/sourceMapModuleConsistent';
import { RequiredTemplateScript } from '../../getters';
import { requireFunction } from '../../requireFunction';

type GenerateContentByParsedTemplateParams = {
  templateParams: TemplateParamsConsistent;
  templateScript: RequiredTemplateScript;
  sourceMap: SourceMapModuleConsistent;
  assets: any;
};

/**
 * Generate content by ParsedTemplateFile, TemplateParams and other detected data
 * */
export const generateContentByParsedTemplate = (
  { parsed, content: originContent }: ParsedTemplateFile,
  { templateParams, ...rest }: GenerateContentByParsedTemplateParams,
) => {
  let resultContent = originContent;
  parsed.forEach(({ originStr, data }) => {
    let currentFragmentResult = '';

    if (data.type === ParsedFragmentDataTypeEnum.variable) {
      currentFragmentResult = templateParams[data.variableName];
    } else if (data.type === ParsedFragmentDataTypeEnum.callFunction) {
      const { functionName, functionArgs } = data;

      // if there is no in "templateParams", then the expression from arg is literal
      const functionArg = templateParams[functionArgs[0]] || functionArgs[0];
      currentFragmentResult = requireFunction({ functionName, functionArg, templateParams, ...rest });
    }

    resultContent = resultContent.replace(originStr, currentFragmentResult);
  });

  return resultContent;
};
