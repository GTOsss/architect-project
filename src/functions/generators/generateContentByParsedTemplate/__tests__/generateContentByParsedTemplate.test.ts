import { parseAllInterpolationMarks } from '../../../parsers';
import { interpolationRegExp } from '../../../parsers/parseTemplateFiles/__tests__/mock';
import { arcMockConfig, arcMockMethods } from '../../../../../__tests__/mock';
import { simpleFunctionTemplate_callFunction, simpleFunctionTemplate_variable } from './mock';
import { generateContentByParsedTemplate } from '../generateContentByParsedTemplate';

const secondArgMock = {
  templateScript: arcMockMethods,
  sourceMap: {},
  assets: [],
};

// todo Refactor for large repeatable block of codes for all file test (need add mock data)

describe('generateContentByParsedTemplate', () => {
  it('simple template with variable', () => {
    const parsedFragments = parseAllInterpolationMarks(
      simpleFunctionTemplate_variable,
      interpolationRegExp,
      arcMockConfig,
    );
    const result = generateContentByParsedTemplate(
      {
        content: simpleFunctionTemplate_variable,
        parsed: parsedFragments,
      },
      {
        templateParams: { name: 'TEST_VALUE_NAME', template: '', outputPath: '' },
        ...secondArgMock,
      },
    );
    expect(result).toMatchSnapshot();
  });

  it('simple template with call function', () => {
    const parsedFragments = parseAllInterpolationMarks(
      simpleFunctionTemplate_callFunction,
      interpolationRegExp,
      arcMockConfig,
    );
    const result = generateContentByParsedTemplate(
      {
        content: simpleFunctionTemplate_callFunction,
        parsed: parsedFragments,
      },
      {
        templateParams: { name: 'TEST_VALUE_NAME', template: '', outputPath: '' },
        ...secondArgMock,
      },
    );
    expect(result).toMatchSnapshot();
  });
});
