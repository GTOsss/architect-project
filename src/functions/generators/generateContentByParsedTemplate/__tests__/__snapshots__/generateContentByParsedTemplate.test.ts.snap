// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateContentByParsedTemplate simple template with call function 1`] = `"const TestValueName = () => {}"`;

exports[`generateContentByParsedTemplate simple template with variable 1`] = `"const TEST_VALUE_NAME = () => {}"`;
